enhancements/network/user-space-pod-iface-and-api-library.md

```diff
@@ -0,0 +1,262 @@
+---
+title: User Space Pod Interface and API Library
+authors:
+  - "@bmcfall"
+reviewers:
+  - "@dcbw"
+  - "@zshi"
+approvers:
+  - TBD
+creation-date: 2020-08-20
+last-updated: 2020-08-20
+status: implementable
+---
+
+# User Space Pod Interface and API Library
+
+Add OCP support for user space applications (i.e. - DPDK based applications)
+running in pods.
+
+## Release Signoff Checklist
+
+- [X] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+A typical pod has one and only one kernel interface (one side of a veth-pair).
+Interface attributes such as IP address and MAC address are applied to the
+interface through the kernel. More advance use-cases involve injecting more than
+one interface into the pod using meta-CNI plugins such as Multus. In this case,
+there are several scenarios that require additional information passed to the
+pod:
+- Multiple interfaces:
+  - With multiple interfaces now in the pod, the pod needs context on which
+  interface is intended for what purpose.
+  - Needed in most multi-interface scenarios, but specifically OCP related,
+  when multiple SR-IOV VFs are injected into a pod.
+- Kernel interface is used by user space application in the pod:
+  - When the container workload performs packet processing in user space,
+  for example a DPDK application, the interface is no longer managed by the
+  kernel in the pod, and data like IP address and MAC address can not use the
```

Comment:
@zshi-redhat Thought I added a comment but missed this one. "we don't have a way to allocate" - What is different on the host/CNI side with regards to allocating an IP when the container is running DPDK?

Reply:
@Billy99  When using sriov-cni for sriov dpdk interface, it only configures the VF attributes such as vlan/qos, but doesn't call IPAM plugin to allocate any IP address. If we want to allocate an IP address for dpdk interface, I think we need the IPAM plugin to support allocating an un-used IP address but not assign it to interface.

Reply:
@zshi-redhat Correct, implementation detail, IPAM is not called. But it could be. Not a technical reason. Currently, network-status has the IPs, but doesn't include the CIDR. I think if we were to push the applying of the IP to the container, we need the CIDR.